services/gateway/endpoints/src/main/java/org/eclipse/ditto/services/gateway/endpoints/directives/ContentTypeValidationDirective.java

```diff
@@ -0,0 +1,111 @@
+  * http://www.eclipse.org/legal/epl-2.0
+  *
+  * SPDX-License-Identifier: EPL-2.0
+  */
+ package org.eclipse.ditto.services.gateway.endpoints.directives;
+
+ import static akka.http.javadsl.server.Directives.complete;
+ import static org.eclipse.ditto.services.gateway.endpoints.utils.DirectivesLoggingUtils.enhanceLogWithCorrelationId;
+
+ import java.text.MessageFormat;
+ import java.util.Optional;
+ import java.util.Set;
+ import java.util.function.Supplier;
+ import java.util.stream.StreamSupport;
+
+ import org.eclipse.ditto.model.base.headers.DittoHeaderDefinition;
+ import org.eclipse.ditto.model.base.headers.DittoHeaders;
+ import org.slf4j.Logger;
+ import org.slf4j.LoggerFactory;
+
+ import akka.http.javadsl.model.HttpHeader;
+ import akka.http.javadsl.model.HttpRequest;
+ import akka.http.javadsl.model.StatusCodes;
+ import akka.http.javadsl.server.RequestContext;
+ import akka.http.javadsl.server.Route;
+
+ /**
+  * Used to validate the Content-Type of a request.
+  */
+ public final class ContentTypeValidationDirective {
+
+     private static final Logger LOGGER = LoggerFactory.getLogger(ContentTypeValidationDirective.class);
+
+     /**
+      * verifies that the content-type of the entity is one of the allowed media-types,
+      * otherwise the request will be completed with 415 ("Unsupported Media Type").
+      *
+      * @param supportedMediaTypes the media-type which are allowed for the wrapped route.
+      * @param ctx the context of the request.
+      * @param dittoHeaders the ditto-headers of a request.
+      * @param inner route to wrap.
+      * @return the wrapped route.
+      */
+     public static Route ensureValidContentType(final Set<String> supportedMediaTypes, final RequestContext ctx,
+             final DittoHeaders dittoHeaders,
+             final Supplier<Route> inner) {
+         return enhanceLogWithCorrelationId(dittoHeaders.getCorrelationId(), () -> {
+             final String requestsMediaType = extractMediaType(ctx.getRequest());
+
+             if(supportedMediaTypes.contains(requestsMediaType)){
+                 return inner.get();
+             } else {
+                 LOGGER.info("Request rejected: unsupported media-type: <{}>  request: <{}>",
+                         requestsMediaType, requestToLogString(ctx.getRequest()));
+                 return completeWithMediaTypeNotSupported(requestsMediaType, supportedMediaTypes);
+             }
+         });
+     }
+
+     protected static Route completeWithMediaTypeNotSupported(final String mediaType,
+             final Set<String> supportedByResource){
+
+             final String msgPatten = "The Media-Type <{0}> is not supported for this endpoint. Allowed " +
+                     "Media-Types are: <{1}>";
+             final String responseMessage =
+                     MessageFormat.format(msgPatten, mediaType, supportedByResource);
+
+             return complete(StatusCodes.UNSUPPORTED_MEDIA_TYPE, responseMessage);
```

Comment:
Shouldn't we return a nice (new) DittoRuntimeException JSON here?

Reply:
Yes and the OpenAPI docs should then also be updated, with the status code and the new error code, right?

Reply:
Hm, unsure. That would be a massive duplication. Documenting this 405 status code seems not to have much value for the api user.
I think it's sufficient to have a good error response message.